cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -63,7 +63,7 @@ func WriteKubeletDynamicEnvFile(cfg *kubeadmapi.ClusterConfiguration, nodeReg *k
 	}
 	stringMap := buildKubeletArgMap(flagOpts)
 	argList := kubeadmutil.BuildArgumentListFromMap(stringMap, nodeReg.KubeletExtraArgs)
-	envFileContent := fmt.Sprintf("%s=%s\n", constants.KubeletEnvFileVariableName, strings.Join(argList, " "))
+	envFileContent := fmt.Sprintf("%s=\"%s\"\n", constants.KubeletEnvFileVariableName, strings.Join(argList, " "))
```

Comment:
i haven't seen this being a problem locally.

Reply:
It would help with the openrc integration, as the initscript can source this file easily on start and make use of the variables.

Reply:
It's OK by me. One tiny nit though, let's use `%q` as it's cleaner that way.

Reply:
Thanks, done!